components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -325,6 +363,63 @@ public void finishFIDO2Registration(String challengeResponse) throws FIDO2Authen
             throw new FIDO2AuthenticatorClientException(message, FIDO2AuthenticatorConstants.ClientExceptionErrorCodes.
                     ERROR_CODE_FINISH_REGISTRATION_INVALID_REQUEST.getErrorCode());
         } else {
+            // Perform webauthn4j attestation validations if enabled.
+            if (getAuthenticatorConfigs().isAttestationValidationEnabled()) {
+                Set<String> transports = response.getCredential().getResponse().getTransports().stream().map(
```

Comment:
Check whether there's a possibility for null pointer exception. In that case handle it
Check other places as well

Reply:
There's no need to add NPE checks as the yubico variables are annotated with `@NonNull`.
For example refer: https://github.com/Yubico/java-webauthn-server/blob/master/webauthn-server-core/src/main/java/com/yubico/webauthn/data/AuthenticatorAttestationResponse.java#L69-L73